File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialog/DialogSplitClips.java
package com.iamkatrechko.clipboardmanager.view.dialog;

import android.app.Activity;
import android.app.Dialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.View;
import android.widget.CheckBox;
import android.widget.EditText;

import com.iamkatrechko.clipboardmanager.R;
import com.iamkatrechko.clipboardmanager.domain.util.UtilPreferences;

/**
 * Диалог объединения заметок в одну
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
public class DialogSplitClips extends DialogFragment {

    /** Текстовое поле с разделитетем */
    private EditText etSplitChar;
    /** Чекбокс необходимости удаления объединенных цитат */
    private CheckBox cbDelete;

    /**
     * Возвращает новый экземпляр диалога
     * @return новый экземпляр диалога
     */
    public static DialogSplitClips newInstance() {
        return new DialogSplitClips();
    }

    /**
     * Возвращает данные обратно во фрагмент
     * @param splitChar      символ-разделитель
     * @param deleteOldClips флаг удаления объединенных заметок
     */
    private void sendResult(String splitChar, boolean deleteOldClips) {
        if (getTargetFragment() == null) {
            return;
        }
        Intent a = new Intent();
        a.putExtra("splitChar", splitChar);
        a.putExtra("deleteOldClips", deleteOldClips);
        getTargetFragment().onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, a);
    }

    @NonNull
    public Dialog onCreateDialog(Bundle bundle) {
        View view = getActivity().getLayoutInflater().inflate(R.layout.dialog_split_clips, null);

        etSplitChar = (EditText) view.findViewById(R.id.etSplitChar);
        cbDelete = (CheckBox) view.findViewById(R.id.cbDelete);

        etSplitChar.setText(UtilPreferences.getSeparator(getActivity()));

        return new AlertDialog.Builder(getActivity())
                .setView(view)
                .setTitle(R.string.title_split_items)
                .setPositiveButton(R.string.split, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialogInterface, int n) {
                        String splitChar = etSplitChar.getText().toString();
                        boolean deleteOldClips = cbDelete.isChecked();

                        UtilPreferences.setSplitChar(getActivity(), splitChar);

                        sendResult(splitChar, deleteOldClips);
                    }
                })
                .setNegativeButton(R.string.cancel, null)
                .create();
    }
}

File: app/src/main/java/com/iamkatrechko/clipboardmanager/view/dialog/DialogSplitClips.kt
package com.iamkatrechko.clipboardmanager.view.dialog

import android.app.Activity
import android.app.Dialog
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.widget.CheckBox
import android.widget.EditText
import com.iamkatrechko.clipboardmanager.R
import com.iamkatrechko.clipboardmanager.domain.util.PrefsManager

/**
 * Диалог объединения заметок в одну
 * @author iamkatrechko
 *         Date: 01.11.2016
 */
class DialogSplitClips : DialogFragment() {

    /**
     * Возвращает данные обратно во фрагмент
     * @param splitChar      символ-разделитель
     * @param deleteOldClips флаг удаления объединенных заметок
     */
    private fun sendResult(splitChar: String, deleteOldClips: Boolean) {
        val intent = Intent().apply {
            putExtra(KEY_SPLIT_CHAR, splitChar)
            putExtra(KEY_IS_DELETE_OLD_CLIPS, deleteOldClips)
        }
        targetFragment?.onActivityResult(targetRequestCode, Activity.RESULT_OK, intent)
    }

    override fun onCreateDialog(bundle: Bundle?): Dialog {
        val view = activity!!.layoutInflater.inflate(R.layout.dialog_split_clips, null)
        val etSplitChar = view.findViewById<EditText>(R.id.etSplitChar)
        val cbDelete = view.findViewById<CheckBox>(R.id.cbDelete)

        etSplitChar.setText(PrefsManager.getInstance().clipSplitChar)

        return AlertDialog.Builder(context!!)
                .setView(view)
                .setTitle(R.string.title_split_items)
                .setPositiveButton(R.string.split) { _, _ ->
                    val splitChar = etSplitChar.text.toString()
                    val deleteOldClips = cbDelete.isChecked

                    PrefsManager.getInstance().clipSplitChar = splitChar
                    sendResult(splitChar, deleteOldClips)
                }
                .setNegativeButton(R.string.cancel, null)
                .create()
    }

    companion object {

        const val KEY_SPLIT_CHAR = "KEY_SPLIT_CHAR"
        const val KEY_IS_DELETE_OLD_CLIPS = "KEY_IS_DELETE_OLD_CLIPS"

        fun newInstance() = DialogSplitClips()
    }
}
